### DISCUSSION TITLE
Whenever langauge is toggled, any existing searchParams in the url dispaear

### DISCUSSION THREAD

**Discussion by brightpixels** [sergiodxa/remix-i18next#77] [Q&A] [answered]
Hello,
Whenever we change the toggle the language the page reloads with the correct language and the url is updated with the relveant searchParams, so something like this www.mydomain.com/page1?lng=es BUT if there is existing searchParams like this www.mydomain.com/page1?currentPage=2&month=3, the  language toggle causes existing seachParams to be lost and replaced with just the lng searchParam.

I have also noticed the same thing happens if the page reloads with errors after form submission. All the existing searchParams are lost, which leads me to think it's not specifically to do with this library but maybe a Remix expected behaviour or bug.

**Comment by sergiodxa** [comment, marked as answer]
This is not remix-i18next issue, when you use a Remix Form component they don’t keep the search params, or if you redirected inside your loader or action you need to keep the search params yourself.

**Reply by brightpixels** [reply]
Thanks for confirming. Do you recommended any particular technique/best practice to persist the search params? I can switch to HTML form instead of Remix Form but this doesn’t deal with the redirect in a loader or action also removes search parents params.
